components/post/components/post-link-preview/lib/GetFirstLink.ts

```diff
@@ -0,0 +1,34 @@
+import { IPost } from '~/models/posts/post/IPost';
+import UrlMatcher from '~/lib/matchers/UrlMatcher';
+
+function isSafeURL(url: string): boolean {
+    try {
+        const parsedURL = new URL(url);
+        const forbiddenURLs = ['localhost', '127.0.0.1', '0.0.0.0'];
```

Comment:
Ipv6 also has localhost on: `::1` & `0000:0000:0000:0000:0000:0000:0000:0001` and 0.0.0.0 is `::/128`.

Reply:
Nice catch! Completely forgot about ipv6 :)